deeplake/auto/structured/dataframe.py

```diff
@@ -20,6 +36,130 @@ def __init__(self, source):
         if not isinstance(self.source, pd.DataFrame):
             raise Exception("Source is not a pandas dataframe object.")
 
+        self.creds = creds
+        self.creds_key = creds_key
+        self._initialize_params(column_params)
+
+    # def _sanitize_tensor(self, input: str):
+    #     """Sanitize a string to be a valid tensor name
+
+    #     Args:
+    #         input (str): A string that will be sanitized
+
+    #     Returns:
+    #         str: A string with the sanitized tensor name
+    #     """
+
+    #     invalid_chars = ["[", "]", "@", ".", ",", "?", "!", "/", "\\", "#", "'", '"']
+    #     new_char = "_"
+    #     for char in invalid_chars:
+    #         input = input.replace(char, new_char)
+
+    #     return input
+
+    def _initialize_params(self, column_params):
+        column_params = column_params if column_params is not None else {}
+
+        keys = list(self.source.columns)
+
+        for key in keys:
+            if key in column_params.keys():
+                column_params[key]["name"] = column_params[key].get(
+                    "name", sanitize_tensor_name(key)
+                )  # Get the specified name, and if it doesn't exist, use the sanitized column name
+            else:
+                column_params[key] = {
+                    "name": sanitize_tensor_name(key)
+                }  # If column parameters are not specified for a column, use the sanitized column name
+
+        self.column_params = column_params
+
+    def _get_most_frequent_image_extension(self, fn_iterator):
+        # TODO: Make this generic and work for any htype that requires compression
+
+        supported_image_extensions = tuple(
+            HTYPE_SUPPORTED_COMPRESSIONS["image"] + ["jpg"]
+        )
+        image_extensions: DefaultDict[str, int] = defaultdict(int)
+
+        for file in fn_iterator:
+            if file.endswith(supported_image_extensions):
+                ext = pathlib.Path(file).suffix[
+                    1:
+                ]  # Get extension without the . symbol
+                image_extensions[ext] += 1
+            else:
+                raise IngestionError(f"The following file is not supported: {file}")
+
+        most_frequent_image_extension = max(
+            image_extensions, key=lambda k: image_extensions[k], default=None
+        )
+
+        return most_frequent_image_extension
```

Comment:
Couple of issues: `endswith` check should include the period (`.`) otherwise a file named `abcdjpg` for example would be counted as a jpeg. Second, creating `pathlib.Path` object in a for loop for basic string operations is not good. Consider refactoring:

```suggestion

    def _get_most_frequent_image_extension(self, fn_iterator):
        # TODO: Make this generic and work for any htype that requires compression
        supported_image_extensions = tuple("." + fmt for fmt in HTYPE_SUPPORTED_COMPRESSIONS["image"] + ["jpg"])
        image_extensions: DefaultDict[str, int] = defaultdict(int)
        for file in fn_iterator:
            if file.lower().endswith(supported_image_extensions):
                ext = file.split(".")[1]
                image_extensions[ext] += 1
            else:
                raise IngestionError(f"The following file is not supported: {file}")

        most_frequent_image_extension = max(
            image_extensions, key=lambda k: image_extensions[k], default=None
        )
        return most_frequent_image_extension
```

Reply:
@ProgerDav Any thoughts here? This applies to all ingestion functions.

Reply:
Yep, I believe this is true for all the ingestion functions. Actually, I guess the "frequent_extensions" parsing part is generally common for all of the ingestions (with minor modifications, though). 

I can try extracting this part into separate utils

Reply:
Hey @farizrahman4u This suggestion breaks cases where there's a period earlier in the path like in the test fixtures. Any idea how to deal with this case without reverting your suggestions.

`'images_basepath': '.test_resources/hub-test-resources/dataframe/images'`

Reply:
@farizrahman4u this `images_basepath` is set by `path = _git_clone(_HUB_TEST_RESOURCES_URL)`